processor.go

```diff
@@ -624,15 +623,26 @@ func (p *CertProcessor) processCertificate(cert Certificate, forMaint bool) (boo
 		return p.NoteCertError(cert, err, "Error while marshalling altNames for domain %v", cert.Spec.Domain)
 	}
 
-	// Save cert details to bolt
-	err = p.db.Update(func(tx *bolt.Tx) error {
-		key := []byte(cert.Spec.Domain)
-		tx.Bucket([]byte("cert-details")).Put(key, certDetailsRaw)
-		tx.Bucket([]byte("domain-altnames")).Put(key, altNamesRaw)
-		return nil
-	})
+	//Need to distinguish if this is a new cert or a renewal.
+	if isRenewal {
+		err = updateCertDetails(cert.Spec.Domain, certDetailsRaw, p.db)
+	} else {
+		err = addCertDetails(cert.Spec.Domain, certDetailsRaw, p.db)
+	}
+
+	if err != nil {
+		return p.NoteCertError(cert, err, "Error while saving certificate data to database for domain %v", cert.Spec.Domain)
+	}
+
+	//Need to distinguish if this is a new cert or a renewal.
+	if isRenewal {
+		err = updateAltNames(cert.Spec.Domain, altNamesRaw, p.db)
+	} else {
+		err = addAltNames(cert.Spec.Domain, altNamesRaw, p.db)
```

Comment:
Same suggestion as CertDetails above.

Reply:
Updated db.go and added a save function.